pkg/staticnetworkconfig/generator.go

```diff
@@ -275,49 +275,97 @@ func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
 	return buf.String(), nil
 }
 
-func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterfaceMap models.MacInterfaceMap, networksConfig []StaticNetworkConfigData) error {
+func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistenceYAML(macInterfaceMap models.MacInterfaceMap, networksYaml, ocpVersion, arch, installerInvoker string) error {
 	interfaceNames := lo.Map(macInterfaceMap, func(m *models.MacInterfaceMapItems0, _ int) string { return m.LogicalNicName })
+
+	var config map[string]interface{}
+
+	// Unmarshal the JSON string into the config struct
+	err := yaml.Unmarshal([]byte(networksYaml), &config)
+	if err != nil {
+		s.log.WithError(err).Errorf("Error unmarshalling JSON")
+		return err
+	}
+
 	// See 'man systemd.net-naming-scheme' for interface naming protocol
 	predicatableIfaceNamePattern := regexp.MustCompile("^en[PsvxXbucaipod]")
-	for _, nc := range networksConfig {
-		cfg, err := ini.LoadSources(ini.LoadOptions{IgnoreInlineComment: true}, []byte(nc.FileContents))
-		if err != nil {
-			s.log.WithError(err).Errorf("Failed to load the ini format string %s", nc.FileContents)
-			return err
+	Interfaces := config["interfaces"].([]interface{})
+	interfaceWithmacIdentifier := make(map[string]struct{})
+
+	for _, iface := range Interfaces {
+		nic := iface.(map[interface{}]interface{})
+		interfaceName, exists := nic["name"]
+		if !exists {
+			return errors.Errorf("interface name not found in networks configuration")
 		}
-		connectionSection := cfg.Section("connection")
-		interfaceName, err := connectionSection.GetKey("interface-name")
-		if err != nil {
-			// When the id field is provided the interface-name will not be present
-			continue
+
+		identifier, exists := nic["identifier"]
+		if exists && identifier == "mac-address" {
+			interfaceWithmacIdentifier[interfaceName.(string)] = struct{}{}
+		}
+	}
+
+	for _, iface := range Interfaces {
+		nic := iface.(map[interface{}]interface{})
+		interfaceName, exists := nic["name"]
+		if !exists {
+			return errors.Errorf("interface name not found in networks configuration")
 		}
-		interfaceType, err := connectionSection.GetKey("type")
+
+		identifier, exists := nic["identifier"]
+		isMacAddressIdentifier := exists && identifier == "mac-address"
+		isVersionOK, err := NMStatectlServiceSupported(ocpVersion, arch)
 		if err != nil {
-			return errors.Wrapf(err, "failed to get interface type for interface %s", interfaceName.String())
+			return err
+		}
+
+		// TODO: This is a temporary validation. It should be removed once the mac-identifier bug in nmstate is resolved.
```

Comment:
I suggest linking to an actual bug report, otherwise nobody in the future is ever going to know if it is resolved or not.